openjdk/src/main/java/org/conscrypt/Platform.java

```diff
@@ -631,6 +636,53 @@ static boolean supportsConscryptCertStore() {
         return false;
     }
 
+    static KeyStore getDefaultCertKeyStore() throws KeyStoreException {
+        // Start with an empty KeyStore.  In the worst case, we'll end up returning it.
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+        try {
+            ks.load(null, null);
+        } catch (CertificateException ignored) {
+        } catch (NoSuchAlgorithmException ignored) {
+        } catch (IOException ignored) {
+            // We're not loading anything, so ignore it
+        }
+        // Find the highest-priority non-Conscrypt provider that provides a PKIX
+        // TrustManagerFactory implementation and ask it for its trusted CAs.  This is most
+        // likely the OpenJDK-provided provider, in which case the platform default properties
+        // for configuring CA certs will be used, but we'll accept any provider that can give
+        // us at least one cert.
+        Provider[] providers = Security.getProviders("TrustManagerFactory.PKIX");
+        for (Provider p : providers) {
+            if (Conscrypt.isConscrypt(p)) {
```

Comment:
why skip conscrypt?

Reply:
@nmittler This is called when we're trying to get a default set of certs for a TrustManager, so if we don't skip Conscrypt, we'll just infinite recurse trying to build a default TrustManager and then find a TrustManager we can get the certs from.

Reply:
That's what I suspected ... mind adding some docs?

Reply:
Sure, done.